src/Microsoft.DocAsCode.Build.Engine/HandlePostProcessors.cs

```diff
@@ -0,0 +1,117 @@
+        }
+
+        public void PrepareMetadata(ImmutableDictionary<string, object> metadata)
+        {
+            foreach (var postProcessor in _postProcessors)
+            {
+                using (new LoggerPhaseScope($"Prepare metadata in post processor {postProcessor.ContractName}", false))
+                using (new PerformanceScope($"Prepare metadata in post processor {postProcessor.ContractName}", LogLevel.Verbose))
+                {
+                    metadata = postProcessor.Processor.PrepareMetadata(metadata);
+                    if (metadata == null)
+                    {
+                        throw new DocfxException($"Plugin {postProcessor.ContractName} should not return null metadata");
+                    }
+                }
+            }
+        }
+
+        public void PostProcess(Manifest manifest, string outputFolder)
+        {
+            // post process
+            foreach (var postProcessor in _postProcessors)
+            {
+                using (new LoggerPhaseScope($"Process in post processor {postProcessor.ContractName}", false))
+                using (new PerformanceScope($"Process in post processor {postProcessor.ContractName}"))
+                {
+                    manifest = postProcessor.Processor.Process(manifest, outputFolder);
+                    if (manifest == null)
+                    {
+                        throw new DocfxException($"Plugin {postProcessor.ContractName} should not return null manifest");
+                    }
+
+                    // To make sure post processor won't generate duplicate output files
+                    ManifestUtility.RemoveDuplicateOutputFiles(manifest.Files);
+                }
+            }
+        }
+
+        private static List<PostProcessor> GetPostProcessor(CompositionHost container, ImmutableArray<string> processors)
+        {
+            var processorList = new List<PostProcessor>();
+            AddBuildInPostProcessor(processorList);
+            foreach (var processor in processors)
+            {
+                var p = CompositionUtility.GetExport<IPostProcessor>(container, processor);
+                if (p != null)
+                {
+                    processorList.Add(new PostProcessor
+                    {
+                        ContractName = processor,
+                        Processor = p
+                    });
+                    Logger.LogInfo($"Post processor {processor} loaded.");
+                }
+                else
+                {
+                    Logger.LogWarning($"Can't find the post processor: {processor}");
+                }
+            }
+            return processorList;
+        }
+
+        private static void AddBuildInPostProcessor(List<PostProcessor> processorList)
+        {
+            processorList.Add(
+                new PostProcessor
+                {
+                    ContractName = "html",
+                    Processor = new HtmlPostProcessor
```

Comment:
is it more natural for HtmlPostProcessor to decide it's ContractName, and it's handlers?